src/main/java/com/microsoft/graph/serializer/ISerializer.java

```diff
@@ -37,25 +43,65 @@
      * @param <T>         the type of the item to be deserialized
      * @return            the deserialized item from the input string
      */
     @Nullable
-    <T> T deserializeObject(@Nonnull final String inputString, @Nonnull final Class<T> clazz);
-    
+    default <T> T deserializeObject(@Nonnull final String inputString, @Nonnull final Class<T> clazz) {
+        return deserializeObject(inputString, clazz, emptyMap());
+    }
+
     /**
      * Deserialize an object from the input string
-     * 
+     *
      * @param inputString     the string that stores the representation of the item
-     * @param clazz           the .class of the item to be deserialized
+     * @param clazz           the {@code Class} of the item to be deserialized
      * @param responseHeaders the HTTP response headers
      * @param <T>             the type of the item to be deserialized
      * @return                the deserialized item from the input string
      */
     @Nullable
     <T> T deserializeObject(@Nonnull final String inputString, @Nonnull final Class<T> clazz, @Nonnull final Map<String, List<String>> responseHeaders);
 
+    /**
+     * Deserialize an object from the input stream
+     *
+     * @param inputStream the stream that stores the representation of the item
+     * @param clazz       the {@code Class} of the item to be deserialized
+     * @param <T>         the type of the item to be deserialized
+     * @return the deserialized item from the input string
+     */
+    @Nullable
+    default <T> T deserializeObject(@Nonnull final InputStream inputStream, @Nonnull final Class<T> clazz) {
+        return deserializeObject(inputStream, clazz, emptyMap());
+    }
+
+    /**
+     * Deserialize an object from the input stream
+     * 
+     * @param inputStream     the stream that stores the representation of the item
+     * @param clazz           the {@code Class} of the item to be deserialized
+     * @param responseHeaders the HTTP response headers
+     * @param <T>             the type of the item to be deserialized
+     * @return                the deserialized item from the input string
+     */
+    @Nullable
+    <T> T deserializeObject(@Nonnull final InputStream inputStream, @Nonnull final Class<T> clazz, @Nonnull final Map<String, List<String>> responseHeaders);
+
+    /**
+     * Deserialize an object from the input JsonElement
+     *
+     * @param jsonElement     the {@code JsonElement} that stores the representation of the item
+     * @param clazz           the {@code Class} of the item to be deserialized
+     * @param responseHeaders the HTTP response headers
+     * @param <T>             the type of the item to be deserialized
+     * @return                the deserialized item from the input string
+     */
+    @SuppressWarnings("unchecked")
+    @Nullable
+    <T> T deserializeObject(@Nonnull JsonElement jsonElement, @Nonnull Class<T> clazz, @Nonnull Map<String, List<String>> responseHeaders);
```

Comment:
can you add a default implementation with an empty map for the the JsonElement override please?

Reply:
Yup, just did that.